### PR TITLE
Add fucntion to detect ivshemme node

### PR DESCRIPTION
**Pull request by yhe39**
Handle the hotplug event to plugout and plugin

### CODE REVIEW

File: drm/ResourceManager.h [hunk 1]
```diff
@@ -63,9 +63,13 @@ class ResourceManager {
   auto GetOrderedConnectors() -> std::vector<DrmConnector *>;
   void UpdateFrontendDisplays();
   void DetachAllFrontendDisplays();
+  void ReloadNode();
 
   std::vector<std::unique_ptr<DrmDevice>> drms_;
 
+  int card_num;
+  int callCount = 0;
```
Comment (feijiang1) [inline drm/ResourceManager.h:71]: callCount is only used in function ResourceManager::UpdateFrontendDisplay, it seems we only need declare it as static varible, don't need declare it as class member. Also callCount should be named as call_count

File: drm/ResourceManager.cpp [hunk 2]
```diff
@@ -105,10 +135,25 @@ void ResourceManager::Init() {
     // is SRI-IOV + dGPU, use virtio-gpu(card2) for display
     if (node_num == 3) {
       std::ostringstream path;
-      path << path_pattern << 2;
+      path << path_pattern << 1;
       auto dev = DrmDevice::CreateInstance(path.str(), this);
```
Comment (feijiang1) [inline drm/ResourceManager.cpp:139]: we will call CreateInstance for both 1 and 2, while only emplace_back for 2, is it expected behavior?

File: drm/ResourceManager.cpp [hunk 3]
```diff
@@ -105,10 +135,25 @@ void ResourceManager::Init() {
...
+      } else {   //is iGPU vf + virtio-gpu + ivshemem, use ivshmem and virtio-gpu
+        if (dev) {
+          drms_.emplace_back(std::move(dev));
+        }
+        std::ostringstream path;
+        path << path_pattern << 2;
+        auto dev = DrmDevice::CreateInstance(path.str(), this);
+        if (dev) {
+          drms_.emplace_back(std::move(dev));
+        }
```
Comment (feijiang1) [inline drm/ResourceManager.cpp:156]: most code is same, can we refine it?

File: drm/ResourceManager.cpp [hunk 4]
```diff
@@ -156,18 +201,30 @@ auto ResourceManager::GetTimeMonotonicNs() -> int64_t {
...
     bool connected = conn->IsConnected();
     bool attached = attached_pipelines_.count(conn) != 0;
 
-    if (connected != attached) {
+    if ((connected != attached) && (hotplug_event_init || boot_event)) {
+
       ALOGI("%s connector %s", connected ? "Attaching" : "Detaching",
             conn->GetName().c_str());
 
+      if (boot_event && (!strcmp(conn->GetName().c_str(), "Virtual-2"))) {
```
Comment (feijiang1) [inline drm/ResourceManager.cpp:220]: Virtual-2 is hard coded here, can we avoid that?

**Review by feijiang1 — COMMENTED** [review]
some comments

**Comment by yhe39** [comment]
hi @feijiang1 , update the patch according your comments:
1. rename the callCount to call_count and define as static var in function.
2. calculate the existing virtual display num and name the cast display as display_cast as the num+1.
3. rewrite the duplicate codes and add comments.
Please help to review. thanks!

**Pull request closed without merging**